tests/libnet/ipaddress.go

```diff
@@ -0,0 +1,33 @@
+package libnet
+
+import (
+	v1 "k8s.io/api/core/v1"
+	netutils "k8s.io/utils/net"
+)
+
+func GetPodIpByFamily(pod *v1.Pod, family v1.IPFamily) string {
+	return getIp(convertPodIpsToStrings(pod.Status.PodIPs), family)
+}
+
+func getIp(ips []string, family v1.IPFamily) string {
+	for _, ip := range ips {
+		if netutils.IsIPv6String(ip) {
+			if family == v1.IPv6Protocol {
+				return ip
+			}
+		} else {
+			if family == v1.IPv4Protocol {
+				return ip
+			}
+		}
+	}
+	return ""
```

Comment:
I prefer to keep it as a `string` since the code that uses `GetPodIpByFamily` is waiting for a `string`. So using a pointer we will compare to `nil` and not an empty string ihich is nice. But we will have to get the pointer string value (which is less nice).